### PR TITLE
Update pygyro to use a more recent version of pyccel

### PR DESCRIPTION
**Pull request by EmilyBourne**
Update pygyro to use a more recent version of pyccel. Remove deprecated external decorator. Make derivative a compulsory argument for spline eval functions.

- Increase minimum pyccel version required
- Remove deprecated external decorator
- Make derivative compulsory (this is the case anyway for numba. If it is not compulsory then the argument must be passed using the keyword to avoid pyccel translating it to an array size, but in this case numba does not see it)
- Use updated flag name
- Update .f90 files generated from most recent pyccel version
- Remove transpose necessary with older version of pyccel. Module can now be accessed directly
- Disable python 3.5 (end of life). Enable python 3.8
- Use codacy for coverage
- Import Cython to fix h5py problems

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,10 +5,11 @@ language: python
 # Enable 3.7 without globally enabling 'dist: xenial' for other build jobs
 matrix:
   include:
-    - python: 3.5
     - python: 3.6
     - python: 3.7
       dist: xenial
+    - python: 3.8
+      dist: xenial
```
Comment (yguclu) [inline .travis.yml:12]: It's better to always use Xenial, which I think has become the default some time ago. See [pyccel](https://github.com/pyccel/pyccel/blob/master/.travis.yml), or maybe even better [sympde](https://github.com/pyccel/sympde/blob/master/.travis.yml).

File: pygyro/advection/__pyccel__/accelerated_advection_steps.f90 [hunk 2]
```diff
@@ -82,314 +80,296 @@ pure subroutine poloidal_advection_step_expl(f, dt, v, rPts, qPts, nPts, &
...
+  !Handle theta boundary conditions
+  !Add the new value of phi to the derivatives
+  !x^{n+1} = x^n + 0.5( f(x^n) + f(x^n + f(x^n)) )
+  !^^^^^^^^^^^^^^^
+  !Step two of Heun method
+  !x^{n+1} = x^n + 0.5( f(x^n) + f(x^n + f(x^n)) )
+  !Clipping is one method of avoiding infinite loops due to
+  !boundary conditions
+  !Using the splines to extrapolate is not sufficient
+  !Find value at the determined point
```
Comment (yguclu) [inline pygyro/advection/__pyccel__/accelerated_advection_steps.f90:143]: It looks like these comments are moved to the wrong place, inside the loop. Do we have an issue for this in Pyccel?

File: requirements.txt [hunk 3]
```diff
@@ -4,7 +4,7 @@ numpy>=1.15.1
 scipy>=1.1.0
 pytest>=2.8.7
 h5py>=2.8.0
-pyccel>=0.9.7
+pyccel>=0.10.0
```
Comment (yguclu) [inline requirements.txt:7]: I think you can remove the note at the end of this file, right?

**Review by yguclu — APPROVED** [review]

**Comment by yguclu** [comment]
@EmilyBourne We can merge this now if you want, but I suggest you make a couple of tiny changes before that...

File: pygyro/advection/__pyccel__/accelerated_advection_steps.f90 (hunk 2 above)
Comment (EmilyBourne) [reply pygyro/advection/__pyccel__/accelerated_advection_steps.f90:143]: I actually hadn't spotted this. The comments come from other functions, it is a coincidence that they are the same comments used in this function. The comments for this function are correctly placed. I have created an issue

**Pull request merged**